xchange-bitstamp/src/main/java/info/bitrich/xchangestream/bitstamp/v2/BitstampStreamingMarketDataService.java

```diff
@@ -0,0 +1,77 @@
+package info.bitrich.xchangestream.bitstamp.v2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.bitstamp.dto.BitstampOrderBook;
+import info.bitrich.xchangestream.bitstamp.dto.BitstampWebSocketTransaction;
+import info.bitrich.xchangestream.core.StreamingMarketDataService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import io.reactivex.Observable;
+import org.knowm.xchange.bitstamp.BitstampAdapters;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Ticker;
+import org.knowm.xchange.dto.marketdata.Trade;
+import org.knowm.xchange.exceptions.NotAvailableFromExchangeException;
+
+import java.util.Date;
+
+/**
+ * Bitstamp WebSocket V2 Streaming Market Data Service implementation
+ * Created by Pavel Chertalev on 15.03.2018.
+ */
+public class BitstampStreamingMarketDataService implements StreamingMarketDataService {
+    private final BitstampStreamingService service;
+
+    public BitstampStreamingMarketDataService(BitstampStreamingService service) {
+        this.service = service;
+    }
+
+    public Observable<OrderBook> getDifferentialOrderBook(CurrencyPair currencyPair, Object... args) {
+        return getOrderBook("diff_order_book", currencyPair, args);
+    }
+
+    @Override
+    public Observable<OrderBook> getOrderBook(CurrencyPair currencyPair, Object... args) {
+        return getOrderBook("order_book", currencyPair, args);
+    }
+
+    private Observable<OrderBook> getOrderBook(String channelPrefix, CurrencyPair currencyPair, Object... args) {
+        String channelName = channelPrefix + getChannelPostfix(currencyPair);
+
+        return service.subscribeChannel(channelName, "data")
+                .map(s -> {
+                    ObjectMapper mapper = StreamingObjectMapperHelper.getObjectMapper();
+                    BitstampOrderBook orderBook = mapper.treeToValue(s.get("data"), BitstampOrderBook.class);
+                    org.knowm.xchange.bitstamp.dto.marketdata.BitstampOrderBook bitstampOrderBook =
```

Comment:
I assume the plan to pull this out of the v1 code when the v1 code is removed?

Reply:
Code a little bit different vs V1, there is used different json node: s.get("data"). But I  believe deprecated V1 code should be removed when old Bitstamp pusher service stopped forever.

Reply:
Makes sense.